### ISSUE TITLE
Files are not deleted

### ISSUE THREAD

**Issue by kokarl** [filesender/filesender#1785]
Hello,

Still on my fresh install filesender 2.44 + SSP 2.1.1
When a user tries to delete a file in an open transfert , I got the following message in the logs : 

[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/autoload.php:193 Autoloader::load(string(SimpleSAML)) Looking for class SimpleSAML, expecting it at /usr/local/www/apache24/data/myfilesender.url/classes/utils/SimpleSAML.class.php but nothing found, may (or may not) be a problem ...

[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/autoload.php:193 Autoloader::load(string(src)) Looking for class src, expecting it at /usr/local/www/apache24/data/myfilesender.url/classes/utils/src.class.php but nothing found, may (or may not) be a problem ...
[2024-02-01 15:49:03] [rest:info] authentication::create(1) saml_uid toto
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/Authentication.class.php:137 DBI::prepare(string(SELECT * FROM Authentications WHERE saml_user_identification_uid = :samluid)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/Authentication.class.php:138 (DBIStatement)->execute(array(:samluid => string(toto))) DBI call
[2024-02-01 15:49:03] [rest:info] authentication::create(2) FOUND AND RETURNING 15
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/User.class.php:282 DBI::prepare(string(SELECT * FROM UserPreferences WHERE authid = :authid)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/User.class.php:283 (DBIStatement)->execute(array(:authid => integer(15))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/User.class.php:210 DBI::prepare(string(SELECT * FROM UserPreferences WHERE id = :id)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/User.class.php:211 (DBIStatement)->execute(array(:id => integer(15))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/www/rest.php:37 RestServer::process() Got "delete" request for endpoint "file/30" with 0 bytes payload
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/www/rest.php:37 RestServer::process() Forwarding call to RestEndpointFile::delete() handler
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/File.class.php:308 DBI::prepare(string(SELECT * FROM Files WHERE id = :id)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/File.class.php:309 (DBIStatement)->execute(array(:id => string(30))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/Transfer.class.php:344 DBI::prepare(string(SELECT * FROM Transfers WHERE id = :id)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/Transfer.class.php:345 (DBIStatement)->execute(array(:id => integer(28))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/File.class.php:403 DBI::prepare(string(SELECT * FROM Files WHERE transfer_id = :transfer_id order by name desc)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/File.class.php:404 (DBIStatement)->execute(array(:transfer_id => integer(28))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:748 DBI::prepare(string(UPDATE Transfers SET userid = :userid, user_email = :user_email, guest_id = :guest_id, lang = :lang, subject = :subject, message = :message, created = :created, made_available = :made_available, expires = :expires, expiry_extensions = :expiry_extensions, status = :status, options = :options, key_version = :key_version, salt = :salt, password_version = :password_version, password_encoding = :password_encoding, password_hash_iterations = :password_hash_iterations, client_entropy = :client_entropy, roundtriptoken = :roundtriptoken, guest_transfer_shown_to_user_who_invited_guest = :guest_transfer_shown_to_user_who_invited_guest WHERE (id = :id))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:749 (DBIStatement)->execute(array(:id => integer(28), :userid => integer(15), :user_email => string(support@mail.com), :guest_id => NULL, :lang => string(en-us), :subject => NULL, :message => NULL, :created => string(2024-02-01 15:48:49), :made_available => string(2024-02-01 15:48:50), :expires => string(2024-02-01 15:49:03), :expiry_extensions => integer(0), :status => string(closed), :options => string({"get_a_link":true,"email_recipient_when_transfer_expires":true,"hide_sender_email":false,"email_me_on_expire":true,"email_upload_complete":true,"email_download_complete":true,"email_daily_statistics":false,"email_report_on_closing":true,"must_be_logged_in_to_download":false,"encryption":false}), :key_version => integer(1), :salt => string(HQVM9xrgvAyfeGZ1cZkCV1mLt51ExHxM), :password_version => integer(1), :password_encoding => integer(0), :password_hash_iterations => integer(4096000), :client_entropy => string(), :roundtriptoken => string(25113b2886d52b2dd4205598e08cb203), :guest_transfer_shown_to_user_who_invited_guest => string(1))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:699 DBI::prepare(string(INSERT INTO AuditLogs(event, target_type, target_id, author_type, author_id, ip, transaction_id, created) VALUES(:event, :target_type, :target_id, :author_type, :author_id, :ip, :transaction_id, :created))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:700 (DBIStatement)->execute(array(:event => string(transfer_closed), :target_type => string(Transfer), :target_id => integer(28), :author_type => string(User), :author_id => integer(15), :ip => string(1.2.3.4), :transaction_id => NULL, :created => string(2024-02-01 15:49:03))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:699 DBI::prepare(string(INSERT INTO StatLogs(event, target_type, size, time_taken, additional_attributes, created, browser, os, is_encrypted) VALUES(:event, :target_type, :size, :time_taken, :additional_attributes, :created, :browser, :os, :is_encrypted))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:700 (DBIStatement)->execute(array(:event => string(transfer_closed), :target_type => string(Transfer), :size => integer(12540), :time_taken => integer(0), :additional_attributes => NULL, :created => string(2024-02-01 15:49:03), :browser => integer(3), :os => integer(9), :is_encrypted => string(0))) DBI call
[2024-02-01 15:49:03] [user 15] [rest:info] Event#transfer_closed on Transfer#28
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack() Array
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack() (
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()     [main] => en-us
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()     [fallback] => Array
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()         (
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()             [0] => en
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()         )
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack() )
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/TranslatableEmail.class.php:210 DBI::prepare(string(SELECT * FROM TranslatableEmails WHERE token = :token)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/TranslatableEmail.class.php:211 (DBIStatement)->execute(array(:token => string(b1d01824-90fc-4731-ab57-8ccb5e9443df))) DBI call
[2024-02-01 15:49:03] [user 15] [rest:info] TranslatableEmail uid generation took 1 tries
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:699 DBI::prepare(string(INSERT INTO TranslatableEmails(context_type, context_id, token, translation_id, variables, created) VALUES(:context_type, :context_id, :token, :translation_id, :variables, :created))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:700 (DBIStatement)->execute(array(:context_type => string(Transfer), :context_id => integer(28), :token => string(b1d01824-90fc-4731-ab57-8ccb5e9443df), :translation_id => string(transfer_deleted_receipt), :variables => string([{"dbobject_type":"User","dbobject_id":15},{"dbobject_type":"Transfer","dbobject_id":28}]), :created => string(2024-02-01 15:49:03))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/RateLimitHistory.class.php:277 DBI::prepare(string(SELECT count(*) as count, 1 as one  FROM RateLimitHistorys WHERE author_context_type = :author_context_type  AND   author_context_id   = :author_context_id  AND   action              = :action  AND   event               = :event  AND   target_context_type = :target_context_type  AND   created >= :created  AND   target_context_id   = :target_context_id )) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/RateLimitHistory.class.php:278 (DBIStatement)->execute(array(:author_context_type => string(User), :author_context_id => integer(15), :action => string(email), :event => string(transfer_deleted_receipt), :target_context_type => string(Transfer), :created => string(2024-01-31 15:0:0), :target_context_id => integer(28))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:699 DBI::prepare(string(INSERT INTO RateLimitHistorys(created, author_context_type, author_context_id, action, event, target_context_type, target_context_id) VALUES(:created, :author_context_type, :author_context_id, :action, :event, :target_context_type, :target_context_id))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:700 (DBIStatement)->execute(array(:created => string(2024-02-01 15:49:03), :author_context_type => string(User), :author_context_id => integer(15), :action => string(email), :event => string(transfer_deleted_receipt), :target_context_type => string(Transfer), :target_context_id => integer(28))) DBI call
[2024-02-01 15:49:03] [user 15] [rest:warn] Sending mail
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:323 DBI::prepare(string(SELECT * FROM AuditLogs  WHERE target_type = :type AND target_id = :id ORDER BY created ASC, id ASC)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:348 (DBIStatement)->execute(array(type => string(Transfer), id => string(28))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/DBI.class.php:315 DBI::prepare(string(SELECT * FROM AuditLogs  WHERE target_type='File' AND (target_id = :ids___0) )) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:348 (DBIStatement)->execute(array(:ids => array)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/Recipient.class.php:215 DBI::prepare(string(SELECT * FROM Recipients WHERE transfer_id = :transfer_id)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/Recipient.class.php:216 (DBIStatement)->execute(array(transfer_id => integer(28))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/DBI.class.php:315 DBI::prepare(string(SELECT * FROM AuditLogs  WHERE target_type='Recipient' AND (target_id = :ids___0) )) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:348 (DBIStatement)->execute(array(:ids => array)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:415 Lang::loadDictionary(string(en_AU)) en_AU
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:415 Lang::loadDictionary(string(en_AU)) /usr/local/www/apache24/data/myfilesender.url/language/en_AU
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:415 Lang::loadDictionary(string(en_AU)) /usr/local/www/apache24/data/myfilesender.url/config/language/en_AU
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:426 Lang::loadDictionary(string(en_AU)) en_AU
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:426 Lang::loadDictionary(string(en_AU)) /usr/local/www/apache24/data/myfilesender.url/language/en_AU
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:426 Lang::loadDictionary(string(en_AU)) /usr/local/www/apache24/data/myfilesender.url/config/language/en_AU
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/TranslatableEmail.class.php:210 DBI::prepare(string(SELECT * FROM TranslatableEmails WHERE token = :token)) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/TranslatableEmail.class.php:211 (DBIStatement)->execute(array(:token => string(12acfbef-df78-422f-be0a-4eab56bde191))) DBI call
[2024-02-01 15:49:03] [user 15] [rest:info] TranslatableEmail uid generation took 1 tries
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:699 DBI::prepare(string(INSERT INTO TranslatableEmails(context_type, context_id, token, translation_id, variables, created) VALUES(:context_type, :context_id, :token, :translation_id, :variables, :created))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:700 (DBIStatement)->execute(array(:context_type => string(Transfer), :context_id => integer(28), :token => string(12acfbef-df78-422f-be0a-4eab56bde191), :translation_id => string(report_inline), :variables => string([{"dbobject_type":"User","dbobject_id":15},{"dbobject_type":"Transfer","dbobject_id":28},{"target":{"type":"Transfer","id":28},"content":{"plain":"Date                  Action that happened\n\n01 Feb 2024 15:48:49  Transfer was created\n\n01 Feb 2024 15:48:49  Upload started\n\n01 Feb 2024 15:48:49  File logo_ciil.png (12.2 kB) uploaded (took 0s)\n\n01 Feb 2024 15:48:50  Upload ended\n\n01 Feb 2024 15:48:50  Transfer became available (took 1s)\n\n01 Feb 2024 15:49:03  Transfer was closed on request\n\n","html":"<tr><td>01 Feb 2024 15:48:49<\/td><td>Transfer was created<\/td><\/tr><tr><td>01 Feb 2024 15:48:49<\/td><td>Upload started<\/td><\/tr><tr><td>01 Feb 2024 15:48:49<\/td><td>File logo_ciil.png (12.2 kB) uploaded (took 0s)<\/td><\/tr><tr><td>01 Feb 2024 15:48:50<\/td><td>Upload ended<\/td><\/tr><tr><td>01 Feb 2024 15:48:50<\/td><td>Transfer became available (took 1s)<\/td><\/tr><tr><td>01 Feb 2024 15:49:03<\/td><td>Transfer was closed on request<\/td><\/tr>"}}]), :created => string(2024-02-01 15:49:03))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/RateLimitHistory.class.php:277 DBI::prepare(string(SELECT count(*) as count, 1 as one  FROM RateLimitHistorys WHERE author_context_type = :author_context_type  AND   author_context_id   = :author_context_id  AND   action              = :action  AND   event               = :event  AND   target_context_type = :target_context_type  AND   created >= :created  AND   target_context_id   = :target_context_id )) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/RateLimitHistory.class.php:278 (DBIStatement)->execute(array(:author_context_type => string(User), :author_context_id => integer(15), :action => string(email), :event => string(report_inline), :target_context_type => string(Transfer), :created => string(2024-01-31 15:0:0), :target_context_id => integer(28))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:699 DBI::prepare(string(INSERT INTO RateLimitHistorys(created, author_context_type, author_context_id, action, event, target_context_type, target_context_id) VALUES(:created, :author_context_type, :author_context_id, :action, :event, :target_context_type, :target_context_id))) DBI call
[2024-02-01 15:49:03] [rest:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:700 (DBIStatement)->execute(array(:created => string(2024-02-01 15:49:03), :author_context_type => string(User), :author_context_id => integer(15), :action => string(email), :event => string(report_inline), :target_context_type => string(Transfer), :target_context_id => integer(28))) DBI call
[2024-02-01 15:49:03] [user 15] [rest:warn] Sending mail
[2024-02-01 15:49:03] [user 15] [rest:error] [StorageFilesystemCannotDeleteException exception uid:65bbaf5ff04e7] storage_filesystem_cannot_delete
[2024-02-01 15:49:03] [user 15] [rest:error] [StorageFilesystemCannotDeleteException trace uid:65bbaf5ff04e7] #0 /usr/local/www/apache24/data/myfilesender.url/classes/storage/Storage.class.php(238): StorageFilesystem::deleteFile()
[2024-02-01 15:49:03] [user 15] [rest:error] [StorageFilesystemCannotDeleteException trace uid:65bbaf5ff04e7] #1 /usr/local/www/apache24/data/myfilesender.url/classes/data/Transfer.class.php(800): Storage::deleteFile()
[2024-02-01 15:49:03] [user 15] [rest:error] [StorageFilesystemCannotDeleteException trace uid:65bbaf5ff04e7] #2 /usr/local/www/apache24/data/myfilesender.url/classes/rest/endpoints/RestEndpointFile.class.php(592): Transfer->close()
[2024-02-01 15:49:03] [user 15] [rest:error] [StorageFilesystemCannotDeleteException trace uid:65bbaf5ff04e7] #3 /usr/local/www/apache24/data/myfilesender.url/classes/rest/RestServer.class.php(299): RestEndpointFile->delete()
[2024-02-01 15:49:03] [user 15] [rest:error] [StorageFilesystemCannotDeleteException trace uid:65bbaf5ff04e7] #4 /usr/local/www/apache24/data/myfilesender.url/www/rest.php(37): RestServer::process()
[2024-02-01 15:49:03] [user 15] [rest:error] [StorageFilesystemCannotDeleteException trace uid:65bbaf5ff04e7] #5 {main}
[2024-02-01 15:49:03] [user 15] [rest:error] [StorageFilesystemCannotDeleteException details uid:65bbaf5ff04e7] path = "/mnt/filesender-data/9823862f-f1ea-4f6e-9cd6-b6779ac53a35"
[2024-02-01 15:49:03] [user 15] [rest:error] [StorageFilesystemCannotDeleteException details uid:65bbaf5ff04e7] file = "File#30(logo.png, 12540 bytes)"
[2024-02-01 15:49:03] [user 15] [rest:error] [StorageFilesystemCannotDeleteException details uid:65bbaf5ff04e7] transfer = "Transfer#28"

The transfert is closed, but the file is never deleted.

Best regards,
Karl

**Comment by WebSpider** [comment]
Cleaning up the files is task of a cronjob. Check scripts/task/cron.php. Usage is documented in the install guide.

**Comment by kokarl** [comment]
cron.php is running well. No error, even in debug mode. Files are not deleted even through the cron.php

cron.php starting up... --force:1 --testing-mode:
cron.php running as user: uid=80(www) gid=80(www) groups=80(www)

cron.php closing expired transfers...
cron.php delete failed transfers...
cron.php close expired guests...
cron.php Delete expired audit logs and related data...
cron.php Send daily summaries...
cron.php Send automatic reminders...
cron.php Report bounces ?...
cron.php Storage warning ?...
cron.php Remove inactive users...
cron.php Clean up old client logs...
cron.php Clean old translated emails...
cron.php Clean old tracking events...
cron.php Clean old tracking events (statlog)...
cron.php Clean old auditlog events...
cron.php Clean old ratelimithistory events...


Logs : 
[2024-02-01 16:40:34] [cron:info] Cron started
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:699 DBI::prepare(string(INSERT INTO StatLogs(event, target_type, size, time_taken, additional_attributes, created, browser, os, is_encrypted) VALUES(:event, :target_type, :size, :time_taken, :additional_attributes, :created, :browser, :os, :is_encrypted))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:700 (DBIStatement)->execute(array(:event => string(global_storage_usage), :target_type => string(global), :size => double(1458402435072), :time_taken => integer(0), :additional_attributes => NULL, :created => string(2024-02-01 16:40:34), :browser => integer(0), :os => integer(0), :is_encrypted => string(0))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:323 DBI::prepare(string(SELECT * FROM UserPreferences  WHERE last_activity >= :date)) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:348 (DBIStatement)->execute(array(:date => string(2024-01-02))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:699 DBI::prepare(string(INSERT INTO StatLogs(event, target_type, size, time_taken, additional_attributes, created, browser, os, is_encrypted) VALUES(:event, :target_type, :size, :time_taken, :additional_attributes, :created, :browser, :os, :is_encrypted))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:700 (DBIStatement)->execute(array(:event => string(global_active_users), :target_type => string(global), :size => integer(2), :time_taken => integer(0), :additional_attributes => NULL, :created => string(2024-02-01 16:40:34), :browser => integer(0), :os => integer(0), :is_encrypted => string(0))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:323 DBI::prepare(string(SELECT * FROM Transfers  WHERE status = 'available' ORDER BY created DESC)) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:348 (DBIStatement)->execute(array()) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:699 DBI::prepare(string(INSERT INTO StatLogs(event, target_type, size, time_taken, additional_attributes, created, browser, os, is_encrypted) VALUES(:event, :target_type, :size, :time_taken, :additional_attributes, :created, :browser, :os, :is_encrypted))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:700 (DBIStatement)->execute(array(:event => string(global_available_transfers), :target_type => string(global), :size => integer(1), :time_taken => integer(0), :additional_attributes => NULL, :created => string(2024-02-01 16:40:34), :browser => integer(0), :os => integer(0), :is_encrypted => string(0))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:323 DBI::prepare(string(SELECT * FROM Transfers  WHERE expires <= :date ORDER BY expires ASC)) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:348 (DBIStatement)->execute(array(:date => string(2024-02-01))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:323 DBI::prepare(string(SELECT * FROM Transfers  WHERE created < :date AND (status = 'created' OR status = 'started' OR status = 'uploading') ORDER BY expires ASC)) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:348 (DBIStatement)->execute(array(:date => string(2024-01-25))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:323 DBI::prepare(string(SELECT * FROM Guests  WHERE expires < :date ORDER BY expires ASC)) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:348 (DBIStatement)->execute(array(:date => string(2024-02-01))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:323 DBI::prepare(string(SELECT * FROM Transfers  WHERE expires <= :date ORDER BY expires ASC)) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:348 (DBIStatement)->execute(array(:date => string(2024-01-01))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:323 DBI::prepare(string(SELECT * FROM Transfers  WHERE status = 'available' ORDER BY created DESC)) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/DBObject.class.php:348 (DBIStatement)->execute(array()) DBI call
[2024-02-01 16:40:34] [cron:info] Storage is warning, reporting
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack() Array
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack() (
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()     [main] => en
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()     [fallback] => Array
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()         (
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()         )
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack() )
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:552 Lang::getCodeStack()
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:426 Lang::loadDictionary(string(en_AU)) en_AU
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:426 Lang::loadDictionary(string(en_AU)) /usr/local/www/apache24/data/myfilesender.url/language/en_AU
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/utils/Lang.class.php:426 Lang::loadDictionary(string(en_AU)) /usr/local/www/apache24/data/myfilesender.url/config/language/en_AU
[2024-02-01 16:40:34] [cron:warn] Sending mail
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/ClientLog.class.php:204 DBI::prepare(string(DELETE FROM ClientLogs WHERE created < :date)) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/ClientLog.class.php:205 (DBIStatement)->execute(array(:date => string(2024-01-22))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/TranslatableEmail.class.php:517 DBI::prepare(string(DELETE FROM TranslatableEmails WHERE created < :date)) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/TranslatableEmail.class.php:518 (DBIStatement)->execute(array(:date => string(2024-01-02))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/TrackingEvent.class.php:323 DBI::prepare(string(DELETE FROM TrackingEvents WHERE created < :date)) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/TrackingEvent.class.php:324 (DBIStatement)->execute(array(:date => string(2023-11-03))) DBI call
[2024-02-01 16:40:34] [cron:info] Cleaning statlogs
[2024-02-01 16:40:34] [cron:info] Statlog keeping set to infinite, nothing to clean
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/AuditLog.class.php:624 DBI::exec(string(update AuditLogs a  set target_id = 'null'  where a.target_type = 'Guest'  and 0=(select count(*) as c from Guests g         where g.id =  cast( a.target_id as unsigned)         limit 1        ) )) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/AuditLog.class.php:630 DBI::exec(string(delete from AuditLogs where target_id = 'null' and target_type = 'Guest' )) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/AuditLog.class.php:638 DBI::prepare(string(delete from AuditLogs where created < :cutoff )) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/AuditLog.class.php:639 (DBIStatement)->execute(array(:cutoff => string(2024-01-01 16:40:34))) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/RateLimitHistory.class.php:310 DBI::prepare(string(delete from RateLimitHistorys where created < :cutoff )) DBI call
[2024-02-01 16:40:34] [cron:debug] /usr/local/www/apache24/data/myfilesender.url/classes/data/RateLimitHistory.class.php:311 (DBIStatement)->execute(array(:cutoff => string(2024-01-01 16:40:34))) DBI call

**Comment by monkeyiq** [comment]
From your original report I see `storage_filesystem_cannot_delete`  being reported. FileSender is looking at path /mnt/filesender-data/9823862f-f1ea-4f6e-9cd6-b6779ac53a35 for file `"File#30(logo.png, 12540 bytes)"`.

If you run a shell as the same user that the filesender cron job is running as can you access this path, if you `cp -av` the  File 30 can you then delete it while running as that user?

**Comment by monkeyiq** [comment]
Sorry, I see that a user is trying to delete the file. So you should instead check if the shell run as the web server uid that is running the FileSender php can access, copy, and delete the file.

**Comment by kokarl** [comment]
I double checked everything about uid and permissions on the files. I can create/copy/delete files in the shell with the account used by Apache to run Filesender.

**Comment by WebSpider** [comment]
> I double checked everything about uid and permissions on the files. I can create/copy/delete files in the shell with the account used by Apache to run Filesender.

If you happen to have selinux enabled, httpd is not allowed to write to parts of the system. By default, I think writing in NFS filesystem for instance, is blocked by default on redhat and redhat-like distributions.

**Comment by monkeyiq** [comment]
If it is selinux related the `getenforce` and `setenforce` can be used to set to a permissive state for quick checking if that is what is blocking things.

**Comment by monkeyiq** [comment]
With the above PR (which is now merged into development) you can set the config option mentioned in the PR to get more information about why the delete has failed. This should show if we are dealing with operation not permitted or shine some light onto why FileSender is having a problem with the delete. I was testing this with the manual delete button in the UI just after uploading the file.

**Comment by kokarl** [comment]
Thanks for your help. My delete problem was due to a bad configuration in my config.php
This config lines prevent filesender to delete files ; 
```
$config['storage_filesystem_tree_deletion_command'] = '/bin/rm -rf';
$config['storage_filesystem_file_deletion_command'] = '/bin/rm';
```
I have to put these lines to work (without the full path on rm command) : 
```
$config['storage_filesystem_tree_deletion_command'] = 'rm -rf';
$config['storage_filesystem_file_deletion_command'] = 'rm';
```
Nevertheless the following line is working as Filesender send me an alert when It lacks space
`$config['storage_filesystem_df_command'] = '/bin/df';`

**Comment by monkeyiq** [comment]
In case somebody else bumps into this issue, what distro are you running? IIRC there is one that doesn't have access to basic tools at `/bin`.

**Comment by monkeyiq** [comment]
It would be handy to know the distro, but I think the issue itself may be resolved now and may be handy if somebody doesnt have the coreutils available at bin in the future.

**Comment by kokarl** [comment]
The "distro", I'm using is the latest 13.X stable release of FreeBSD. All the basic tools are in /bin, like Linux distros
